### PR TITLE
WW-4818 change default Multipart validation regex to comply with RFC1341

### PR DESCRIPTION
**Pull request by sdutry**

### CODE REVIEW

File: core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java [hunk 1]
```diff
@@ -88,7 +88,7 @@
      */
     public static final String REQUEST_POST_METHOD = "POST";
 
-    public static final String MULTIPART_FORM_DATA_REGEX = "^multipart\\/form-data(; boundary=[\\-a-zA-Z0-9]{1,70})?";
+    public static final String MULTIPART_FORM_DATA_REGEX = "^multipart/form-data(; boundary=[0-9a-zA-Z'()+_,\\-./:=?]{1,70})?";
```
Comment (lukaszlenart) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: You have to escape `.` as it now means `Any single character`

Comment (lukaszlenart) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: Also `-` has to be escaped as it is treated as a group separator

Comment (sdutry) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: not within the square brackets according to my knowledge

Comment (lukaszlenart) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: o! didn't know that :)

Comment (sdutry) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: you are correct about the `-` needing to be escaped , but that's the only one i escaped. (they're in the order they're mentioned in RFC1341 (so i moved the `-` backwards in the pattern)

Comment (lukaszlenart) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: There are two test cases that you can extend or another one - see `DispatcherTest`

Comment (sdutry) [inline core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java]: Ok, i'll add some tests to confirm the pattern is correct.
(Or if needed to correct it)

**Comment by sdutry** [comment]
@lukaszlenart 
I added 2 simple tests.
- one containing all the special allowed characters
- another one containing a single not-allowed character

Please feel free to tell me what other test-cases you want added.
(for example, any specific characters you want tested?)

**Comment by lukaszlenart** [comment]
Looks good 👍  LGTM :)

**Comment by sdutry** [comment]
@lukaszlenart 
Am i allowed to merge this or is there more work/checks that needs to happen first?

**Comment by lukaszlenart** [comment]
@sdutry yes, you are :) 
http://struts.apache.org/submitting-patches.html#how-to-merge-pull-requests

**Pull request merged**